src/libraries/Microsoft.PowerFx.Core/Texl/BuiltinFunctionsCore.cs

```diff
@@ -23,243 +23,709 @@ internal class BuiltinFunctionsCore
+            ForAll_UO,
+            GUID_UO,
+            GUIDNoArg,
+            GUIDPure,
+            Hex2Dec,
+            Hex2DecT,
+            Hour,
+            If,
+            IfError,
+            Index,
+            Index_UO,
+            Int,
+            IntT,
+            IsBlank,
+            IsBlankOptionSetValue,
+            IsBlankOrError,
+            IsBlankOrErrorOptionSetValue,
+            IsEmpty,
+            IsError,
+            IsNumeric,
+            ISOWeekNum,
+            IsToday,
+            Language,
+            Last,
+            Last_UO,
+            LastN,
+            LastN_UO,
+            Left,
+            LeftST,
+            LeftTS,
+            LeftTT,
+            Len,
+            LenT,
+            Ln,
+            LnT,
+            Log,
+            LogT,
+            LookUp,
+            Lower,
+            LowerT,
+            Max,
+            MaxT,
+            Mid,
+            MidT,
+            Min,
+            MinT,
+            Minute,
+            Mod,
+            ModT,
+            Month,
+            MonthsLong,
+            MonthsShort,
+            Not,
+            Now,
+            Or,
+            ParseJSON,
+            PatchRecord,
+            Pi,
+            PlainText,
+            Power,
+            PowerT,
+            Proper,
+            ProperT,
+            Radians,
+            RadiansT,
+            Rand,
+            RandBetween,
+            Refresh,
+            RenameColumns,
+            Replace,
+            ReplaceT,
+            RGBA,
+            Right,
+            RightST,
+            RightTS,
+            RightTT,
+            Round,
+            RoundDown,
+            RoundDownT,
+            RoundT,
+            RoundUp,
+            RoundUpT,
+            Search,
+            Second,
+            Sequence,
+            ShowColumns,
+            Shuffle,
+            Sin,
+            SinT,
+            Sort,
+            SortByColumns,
+            Split,
+            Sqrt,
+            SqrtT,
+            StartsWith,
+            StdevP,
+            StdevPT,
+            Substitute,
+            SubstituteT,
+            Sum,
+            SumT,
+            Switch,
+            Table,
+            Table_UO,
+            Tan,
+            TanT,
+            Text,
+            Text_UO,
+            Time,
+            TimeValue,
+            TimeValue_UO,
+            TimeZoneOffset,
+            Today,
+            Trace,
+            Trim,
+            TrimEnds,
+            TrimEndsT,
+            TrimT,
+            Trunc,
+            TruncT,
+            UniChar,
+            UniCharT,
+            Upper,
+            UpperT,
+            Value,
+            Value_UO,
+            VarP,
+            VarPT,
+            Weekday,
+            WeekdaysLong,
+            WeekdaysShort,
+            WeekNum,
+            With,
+            Year
+        }
+
+        public static readonly TexlFunction Abs = _functionArray[(int)Library.Abs];

```

Comment:
>(int)Library.Abs [](http://example.com/codeflow?start=65&length=16)

This is dangerous, as if it assumes that the enum and the arrays will be kept in sync. If we want to reference those in this syntax, _functionArray should be a _functionDictionary<Library, TexlFunction> instead.

Reply:
although we could mitigate that concern with a unit test?

Dictionary is good too, but then Abs is a dictionary lookup, whereas just a direct reference to the texlFunction.

Reply:
I'm not sure there is any real better solution
I've used a dictionary